{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "357c9bbd",
   "metadata": {},
   "source": [
    "# Retrieving metabolite information\n",
    "\n",
    "The addition of metabolite related information to biomedical knowledge graphs is gaining importance to describe and learn about genome druggability and disease pathology for several applications such as drug repurposing or disease biomarker discovery. In this BioHackathon, we have investigated how to retrieve metabolite associations from curated databases focusing on the specific use case of drug repurposing on inborn errors of metabolism disorders.\n",
    "\n",
    "(uniprot)=\n",
    "## UniProt\n",
    "\n",
    "Here is a link [to the intro](intro.md). Here is the link to [](uniprot) section. UniProt is a comprehensive knowledge base that focuses its metabolite content on natural products.\n",
    "\n",
    "[Example SPARQL notebook for SIB](https://github.com/biosoda/tutorial_orthology/blob/master/Orthology_SPARQL_Notebook.ipynb)\n",
    "\n",
    "[Another example of SPARQL notebook for BIND](https://github.com/LUMC-BioSemantics/bind/blob/main/queries/bind_wp_queries_model_v1.ipynb)\n",
    "\n",
    "### Exploring genome druggability\n",
    "\n",
    "\n",
    "#### Retrieving drug candidates based on activity metabolite similarity (UniProt, Rhea, ...)\n",
    "    1. In UniProt, query for catalytic reactions (#39 + 40 at https://sparql.uniprot.org/.well-known/sparql-examples/) \n",
    "First, retrieve all enzymes. Then the catalytic reaction and its accession ID. Then filter human enzymes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "381fe514",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: SPARQLWrapper in /home/nur/anaconda3/lib/python3.10/site-packages (2.0.0)\r\n",
      "Requirement already satisfied: rdflib>=6.1.1 in /home/nur/anaconda3/lib/python3.10/site-packages (from SPARQLWrapper) (6.3.2)\r\n",
      "Requirement already satisfied: pyparsing<4,>=2.1.0 in /home/nur/anaconda3/lib/python3.10/site-packages (from rdflib>=6.1.1->SPARQLWrapper) (3.0.9)\r\n",
      "Requirement already satisfied: isodate<0.7.0,>=0.6.0 in /home/nur/anaconda3/lib/python3.10/site-packages (from rdflib>=6.1.1->SPARQLWrapper) (0.6.1)\r\n",
      "Requirement already satisfied: six in /home/nur/anaconda3/lib/python3.10/site-packages (from isodate<0.7.0,>=0.6.0->rdflib>=6.1.1->SPARQLWrapper) (1.16.0)\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_100073/2559832190.py:8: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.\n",
      "  pd.set_option('display.max_colwidth', -1)\n"
     ]
    }
   ],
   "source": [
    "# first imports to use SPARQL endpoints of each source and execute SPARQL queries\n",
    "import sys, os, time\n",
    "!{sys.executable} -m pip install SPARQLWrapper\n",
    "from SPARQLWrapper import SPARQLWrapper, JSON\n",
    "import pandas as pd\n",
    "\n",
    "# always display full column results (don't truncate output)\n",
    "pd.set_option('display.max_colwidth', -1)\n",
    "\n",
    "# define the endpoints as wrappers for executing SPARQL queries\n",
    "sparql_uniprot = SPARQLWrapper(\"https://sparql.uniprot.org/sparql\")\n",
    "\n",
    "# function to print in a table results of a SPARQL query\n",
    "def pretty_print(results):\n",
    "    \n",
    "    # how to transform SPARQL results into Pandas dataframes\n",
    "    \n",
    "    # get header (column names) from results\n",
    "    header = results[\"results\"][\"bindings\"][0].keys()\n",
    "\n",
    "    # display table of results:\n",
    "    table = []\n",
    "    \n",
    "    # the SPARQL JSON results to the query are available in the \"results\", \"bindings\" entry:\n",
    "    for entry in results[\"results\"][\"bindings\"]:\n",
    "        # append entries from the results to a regular Python list of rows, which we can then transform to a Pandas DF\n",
    "        row = [entry[column][\"value\"] if entry.get(column, None) != None else None for column in header]\n",
    "        table.append(row)\n",
    "    df = pd.DataFrame(table, columns=list(header))\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "26935b98",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Queries in UniProt-RDF\n",
    "# Query for human enzymes and retrieve the catalytic reaction(s) IDs in Rhea\n",
    "\n",
    "query_human_enzymes = \"\"\"\n",
    "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n",
    "PREFIX wikibase: <http://wikiba.se/ontology#>\n",
    "PREFIX wdt: <http://www.wikidata.org/prop/direct/>\n",
    "PREFIX wd: <http://www.wikidata.org/entity/>\n",
    "PREFIX vg: <http://biohackathon.org/resource/vg#>\n",
    "PREFIX up: <http://purl.uniprot.org/core/>\n",
    "PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>\n",
    "PREFIX uberon: <http://purl.obolibrary.org/obo/uo#>\n",
    "PREFIX taxon: <http://purl.uniprot.org/taxonomy/>\n",
    "PREFIX sp: <http://spinrdf.org/sp#>\n",
    "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>\n",
    "PREFIX sio: <http://semanticscience.org/resource/>\n",
    "PREFIX sh: <http://www.w3.org/ns/shacl#>\n",
    "PREFIX schema: <http://schema.org/>\n",
    "PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>\n",
    "PREFIX rh: <http://rdf.rhea-db.org/>\n",
    "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>\n",
    "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "PREFIX pubmed: <http://rdf.ncbi.nlm.nih.gov/pubmed/>\n",
    "PREFIX ps: <http://www.wikidata.org/prop/statement/>\n",
    "PREFIX pq: <http://www.wikidata.org/prop/qualifier/>\n",
    "PREFIX patent: <http://data.epo.org/linked-data/def/patent/>\n",
    "PREFIX p: <http://www.wikidata.org/prop/>\n",
    "PREFIX owl: <http://www.w3.org/2002/07/owl#>\n",
    "PREFIX orthodbGroup: <http://purl.orthodb.org/odbgroup/>\n",
    "PREFIX orthodb: <http://purl.orthodb.org/>\n",
    "PREFIX orth: <http://purl.org/net/orth#>\n",
    "PREFIX obo: <http://purl.obolibrary.org/obo/>\n",
    "PREFIX np: <http://nextprot.org/rdf#>\n",
    "PREFIX nextprot: <http://nextprot.org/rdf/entry/>\n",
    "PREFIX mnx: <https://rdf.metanetx.org/schema/>\n",
    "PREFIX mnet: <https://rdf.metanetx.org/mnet/>\n",
    "PREFIX mesh: <http://id.nlm.nih.gov/mesh/>\n",
    "PREFIX lscr: <http://purl.org/lscr#>\n",
    "PREFIX lipidmaps: <https://www.lipidmaps.org/rdf/>\n",
    "PREFIX keywords: <http://purl.uniprot.org/keywords/>\n",
    "PREFIX insdcschema: <http://ddbj.nig.ac.jp/ontologies/nucleotide/>\n",
    "PREFIX insdc: <http://identifiers.org/insdc/>\n",
    "PREFIX identifiers: <http://identifiers.org/>\n",
    "PREFIX glyconnect: <https://purl.org/glyconnect/>\n",
    "PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>\n",
    "PREFIX genex: <http://purl.org/genex#>\n",
    "PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n",
    "PREFIX faldo: <http://biohackathon.org/resource/faldo#>\n",
    "PREFIX eunisSpecies: <http://eunis.eea.europa.eu/rdf/species-schema.rdf#>\n",
    "PREFIX ensembltranscript: <http://rdf.ebi.ac.uk/resource/ensembl.transcript/>\n",
    "PREFIX ensemblterms: <http://rdf.ebi.ac.uk/terms/ensembl/>\n",
    "PREFIX ensemblprotein: <http://rdf.ebi.ac.uk/resource/ensembl.protein/>\n",
    "PREFIX ensemblexon: <http://rdf.ebi.ac.uk/resource/ensembl.exon/>\n",
    "PREFIX ensembl: <http://rdf.ebi.ac.uk/resource/ensembl/>\n",
    "PREFIX ec: <http://purl.uniprot.org/enzyme/>\n",
    "PREFIX dcterms: <http://purl.org/dc/terms/>\n",
    "PREFIX dc: <http://purl.org/dc/terms/>\n",
    "PREFIX chebislash: <http://purl.obolibrary.org/obo/chebi/>\n",
    "PREFIX chebihash: <http://purl.obolibrary.org/obo/chebi#>\n",
    "PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>\n",
    "PREFIX busco: <http://busco.ezlab.org/schema#>\n",
    "PREFIX bibo: <http://purl.org/ontology/bibo/>\n",
    "PREFIX allie: <http://allie.dbcls.jp/>\n",
    "PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>\n",
    "PREFIX GO: <http://purl.obolibrary.org/obo/GO_>\n",
    "PREFIX ECO: <http://purl.obolibrary.org/obo/ECO_>\n",
    "PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>\n",
    "SELECT DISTINCT \n",
    "  ?protein \n",
    "  ?rhea\n",
    "WHERE {\n",
    "  ?protein up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea ;\n",
    "    up:organism taxon:9606 .\n",
    "}\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "50e7bf99",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 6\u001b[0m\n\u001b[1;32m      3\u001b[0m sparql_uniprot\u001b[38;5;241m.\u001b[39msetReturnFormat(JSON)\n\u001b[1;32m      5\u001b[0m NUM_EXAMPLES\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m3\u001b[39m\n\u001b[0;32m----> 6\u001b[0m results_human_enzymes \u001b[38;5;241m=\u001b[39m \u001b[43msparql_uniprot\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquery\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconvert\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      7\u001b[0m pretty_print(results_human_enzymes)\u001b[38;5;241m.\u001b[39mhead(NUM_EXAMPLES)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/SPARQLWrapper/Wrapper.py:1196\u001b[0m, in \u001b[0;36mQueryResult.convert\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1194\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m _content_type_in_list(ct, _SPARQL_JSON):\n\u001b[1;32m   1195\u001b[0m     _validate_format(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mJSON\u001b[39m\u001b[38;5;124m\"\u001b[39m, [JSON], ct, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrequestedFormat)\n\u001b[0;32m-> 1196\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_convertJSON\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1197\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m _content_type_in_list(ct, _RDF_XML):\n\u001b[1;32m   1198\u001b[0m     _validate_format(\n\u001b[1;32m   1199\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRDF/XML\u001b[39m\u001b[38;5;124m\"\u001b[39m, [RDF, XML, RDFXML], ct, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrequestedFormat\n\u001b[1;32m   1200\u001b[0m     )\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/site-packages/SPARQLWrapper/Wrapper.py:1059\u001b[0m, in \u001b[0;36mQueryResult._convertJSON\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1051\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_convertJSON\u001b[39m(\u001b[38;5;28mself\u001b[39m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Dict[Any, Any]:\n\u001b[1;32m   1052\u001b[0m     \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   1053\u001b[0m \u001b[38;5;124;03m    Convert a JSON result into a Python dict. This method can be overwritten in a subclass\u001b[39;00m\n\u001b[1;32m   1054\u001b[0m \u001b[38;5;124;03m    for a different conversion method.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1057\u001b[0m \u001b[38;5;124;03m    :rtype: dict\u001b[39;00m\n\u001b[1;32m   1058\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 1059\u001b[0m     json_str \u001b[38;5;241m=\u001b[39m json\u001b[38;5;241m.\u001b[39mloads(\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mresponse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mdecode(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mutf-8\u001b[39m\u001b[38;5;124m\"\u001b[39m))\n\u001b[1;32m   1060\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(json_str, \u001b[38;5;28mdict\u001b[39m):\n\u001b[1;32m   1061\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m json_str\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/http/client.py:460\u001b[0m, in \u001b[0;36mHTTPResponse.read\u001b[0;34m(self, amt)\u001b[0m\n\u001b[1;32m    457\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    459\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mchunked:\n\u001b[0;32m--> 460\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_read_chunked\u001b[49m\u001b[43m(\u001b[49m\u001b[43mamt\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    462\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m amt \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    463\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mlength \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m amt \u001b[38;5;241m>\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mlength:\n\u001b[1;32m    464\u001b[0m         \u001b[38;5;66;03m# clip the read to the \"end of response\"\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/http/client.py:583\u001b[0m, in \u001b[0;36mHTTPResponse._read_chunked\u001b[0;34m(self, amt)\u001b[0m\n\u001b[1;32m    581\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    582\u001b[0m     \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m--> 583\u001b[0m         chunk_left \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_chunk_left\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    584\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m chunk_left \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    585\u001b[0m             \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/http/client.py:566\u001b[0m, in \u001b[0;36mHTTPResponse._get_chunk_left\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    564\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_safe_read(\u001b[38;5;241m2\u001b[39m)  \u001b[38;5;66;03m# toss the CRLF at the end of the chunk\u001b[39;00m\n\u001b[1;32m    565\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 566\u001b[0m     chunk_left \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_read_next_chunk_size\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    567\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m:\n\u001b[1;32m    568\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m IncompleteRead(\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/http/client.py:526\u001b[0m, in \u001b[0;36mHTTPResponse._read_next_chunk_size\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    524\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_read_next_chunk_size\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[1;32m    525\u001b[0m     \u001b[38;5;66;03m# Read the next chunk size from the file\u001b[39;00m\n\u001b[0;32m--> 526\u001b[0m     line \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreadline\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_MAXLINE\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m    527\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(line) \u001b[38;5;241m>\u001b[39m _MAXLINE:\n\u001b[1;32m    528\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m LineTooLong(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mchunk size\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/socket.py:705\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    703\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m    704\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 705\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecv_into\u001b[49m\u001b[43m(\u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    706\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m timeout:\n\u001b[1;32m    707\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_timeout_occurred \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/ssl.py:1307\u001b[0m, in \u001b[0;36mSSLSocket.recv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1303\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m flags \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m   1304\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m   1305\u001b[0m           \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnon-zero flags not allowed in calls to recv_into() on \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m\n\u001b[1;32m   1306\u001b[0m           \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m)\n\u001b[0;32m-> 1307\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnbytes\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbuffer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1308\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1309\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39mrecv_into(buffer, nbytes, flags)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.10/ssl.py:1163\u001b[0m, in \u001b[0;36mSSLSocket.read\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1161\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m   1162\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m buffer \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m-> 1163\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sslobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbuffer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1164\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1165\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;28mlen\u001b[39m)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# set the query to be executed against the UniProt endpoint and set the return format to JSON\n",
    "sparql_uniprot.setQuery(query_human_enzymes)\n",
    "sparql_uniprot.setReturnFormat(JSON)\n",
    "\n",
    "NUM_EXAMPLES=3\n",
    "results_human_enzymes = sparql_uniprot.query().convert()\n",
    "pretty_print(results_human_enzymes).head(NUM_EXAMPLES)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37db4fa6",
   "metadata": {},
   "source": [
    "    2. In RHEA, query for metabolites participating in catalytic reactions (#13)\n",
    "With this information query RHEA to retrieve reaction participants involved in metabolic pathways, and get the ChEBI IDs from both reactants and products. Then cross IDSM Sachem with these identifiers to expand these known compounds in ChEBI with similar compounds in PubChem. In PubChem we can retrieve three types of compounds: 1. known drugs that have protein targets with which have activity (useful for link prediction); 2. ChEMBL targets with known sensibility; 3. compounds with unknown relevance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bfc5e77a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Queries in UniProt-RDF\n",
    "# Query for human enzymes and retrieve the catalytic reaction(s) IDs in Rhea\n",
    "# Query Rhea for metabolites participating in catalytic reactions\n",
    "\n",
    "query_metabolites = \"\"\"\n",
    "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n",
    "PREFIX wikibase: <http://wikiba.se/ontology#>\n",
    "PREFIX wdt: <http://www.wikidata.org/prop/direct/>\n",
    "PREFIX wd: <http://www.wikidata.org/entity/>\n",
    "PREFIX vg: <http://biohackathon.org/resource/vg#>\n",
    "PREFIX up: <http://purl.uniprot.org/core/>\n",
    "PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>\n",
    "PREFIX uberon: <http://purl.obolibrary.org/obo/uo#>\n",
    "PREFIX taxon: <http://purl.uniprot.org/taxonomy/>\n",
    "PREFIX sp: <http://spinrdf.org/sp#>\n",
    "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>\n",
    "PREFIX sio: <http://semanticscience.org/resource/>\n",
    "PREFIX sh: <http://www.w3.org/ns/shacl#>\n",
    "PREFIX schema: <http://schema.org/>\n",
    "PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>\n",
    "PREFIX rh: <http://rdf.rhea-db.org/>\n",
    "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>\n",
    "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "PREFIX pubmed: <http://rdf.ncbi.nlm.nih.gov/pubmed/>\n",
    "PREFIX ps: <http://www.wikidata.org/prop/statement/>\n",
    "PREFIX pq: <http://www.wikidata.org/prop/qualifier/>\n",
    "PREFIX patent: <http://data.epo.org/linked-data/def/patent/>\n",
    "PREFIX p: <http://www.wikidata.org/prop/>\n",
    "PREFIX owl: <http://www.w3.org/2002/07/owl#>\n",
    "PREFIX orthodbGroup: <http://purl.orthodb.org/odbgroup/>\n",
    "PREFIX orthodb: <http://purl.orthodb.org/>\n",
    "PREFIX orth: <http://purl.org/net/orth#>\n",
    "PREFIX obo: <http://purl.obolibrary.org/obo/>\n",
    "PREFIX np: <http://nextprot.org/rdf#>\n",
    "PREFIX nextprot: <http://nextprot.org/rdf/entry/>\n",
    "PREFIX mnx: <https://rdf.metanetx.org/schema/>\n",
    "PREFIX mnet: <https://rdf.metanetx.org/mnet/>\n",
    "PREFIX mesh: <http://id.nlm.nih.gov/mesh/>\n",
    "PREFIX lscr: <http://purl.org/lscr#>\n",
    "PREFIX lipidmaps: <https://www.lipidmaps.org/rdf/>\n",
    "PREFIX keywords: <http://purl.uniprot.org/keywords/>\n",
    "PREFIX insdcschema: <http://ddbj.nig.ac.jp/ontologies/nucleotide/>\n",
    "PREFIX insdc: <http://identifiers.org/insdc/>\n",
    "PREFIX identifiers: <http://identifiers.org/>\n",
    "PREFIX glyconnect: <https://purl.org/glyconnect/>\n",
    "PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>\n",
    "PREFIX genex: <http://purl.org/genex#>\n",
    "PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n",
    "PREFIX faldo: <http://biohackathon.org/resource/faldo#>\n",
    "PREFIX eunisSpecies: <http://eunis.eea.europa.eu/rdf/species-schema.rdf#>\n",
    "PREFIX ensembltranscript: <http://rdf.ebi.ac.uk/resource/ensembl.transcript/>\n",
    "PREFIX ensemblterms: <http://rdf.ebi.ac.uk/terms/ensembl/>\n",
    "PREFIX ensemblprotein: <http://rdf.ebi.ac.uk/resource/ensembl.protein/>\n",
    "PREFIX ensemblexon: <http://rdf.ebi.ac.uk/resource/ensembl.exon/>\n",
    "PREFIX ensembl: <http://rdf.ebi.ac.uk/resource/ensembl/>\n",
    "PREFIX ec: <http://purl.uniprot.org/enzyme/>\n",
    "PREFIX dcterms: <http://purl.org/dc/terms/>\n",
    "PREFIX dc: <http://purl.org/dc/terms/>\n",
    "PREFIX chebislash: <http://purl.obolibrary.org/obo/chebi/>\n",
    "PREFIX chebihash: <http://purl.obolibrary.org/obo/chebi#>\n",
    "PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>\n",
    "PREFIX busco: <http://busco.ezlab.org/schema#>\n",
    "PREFIX bibo: <http://purl.org/ontology/bibo/>\n",
    "PREFIX allie: <http://allie.dbcls.jp/>\n",
    "PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>\n",
    "PREFIX GO: <http://purl.obolibrary.org/obo/GO_>\n",
    "PREFIX ECO: <http://purl.obolibrary.org/obo/ECO_>\n",
    "PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>\n",
    "SELECT DISTINCT \n",
    "  ?protein \n",
    "  ?rhea\n",
    "  ?chebi\n",
    "WHERE {\n",
    "  ?protein up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea ;\n",
    "    up:organism taxon:9606 .\n",
    "  # Query Rhea for   \n",
    "  SERVICE <https://sparql.rhea-db.org/sparql> {\n",
    "        ?rhea rh:side/rh:contains/rh:compound/rdfs:subClassOf ?chebi .\n",
    "      }\n",
    "}\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3eedffdf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chebi</th>\n",
       "      <th>protein</th>\n",
       "      <th>rhea</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://rdf.rhea-db.org/SmallMolecule</td>\n",
       "      <td>http://purl.uniprot.org/uniprot/B2RB89</td>\n",
       "      <td>http://rdf.rhea-db.org/17989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://purl.obolibrary.org/obo/CHEBI_30616</td>\n",
       "      <td>http://purl.uniprot.org/uniprot/B2RB89</td>\n",
       "      <td>http://rdf.rhea-db.org/17989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://rdf.rhea-db.org/GenericPolypeptide</td>\n",
       "      <td>http://purl.uniprot.org/uniprot/B2RB89</td>\n",
       "      <td>http://rdf.rhea-db.org/17989</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        chebi  \\\n",
       "0  http://rdf.rhea-db.org/SmallMolecule         \n",
       "1  http://purl.obolibrary.org/obo/CHEBI_30616   \n",
       "2  http://rdf.rhea-db.org/GenericPolypeptide    \n",
       "\n",
       "                                  protein                          rhea  \n",
       "0  http://purl.uniprot.org/uniprot/B2RB89  http://rdf.rhea-db.org/17989  \n",
       "1  http://purl.uniprot.org/uniprot/B2RB89  http://rdf.rhea-db.org/17989  \n",
       "2  http://purl.uniprot.org/uniprot/B2RB89  http://rdf.rhea-db.org/17989  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# set the query to be executed against the UniProt endpoint and set the return format to JSON\n",
    "sparql_uniprot.setQuery(query_metabolites)\n",
    "sparql_uniprot.setReturnFormat(JSON)\n",
    "\n",
    "NUM_EXAMPLES=3\n",
    "results_metabolites = sparql_uniprot.query().convert()\n",
    "pretty_print(results_metabolites).head(NUM_EXAMPLES)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58dc6f26",
   "metadata": {},
   "source": [
    "#### Retrieving drug candidates based on structural metabolite similariy\n",
    "From our user metabolite drug candidates, retrieve similar ligands in PDB."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fbd8fb7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13b5decc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "75392052",
   "metadata": {},
   "source": [
    "#### Explore DBCLS databases for ligands and druggability."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d536598",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61cc95f8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c85e22a4",
   "metadata": {},
   "source": [
    "### Exploring disease pathology\n",
    "\n",
    "#### Retrieval of curated disease-metabolite associations\n",
    "From IMCD, retrieve a curated dataset of metabolite-disease associations. There is other the Recon models (Leiden) data source rich on this metabolite content. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a42b4fe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30e02688",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "49d9bb02",
   "metadata": {},
   "source": [
    "#### Retrieval of isoforms\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d32bcc7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a60e53b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0c6d743a",
   "metadata": {},
   "source": [
    "#### Retrieval of disease enzymes that interact with Cholestane metabolite substructure\n",
    "\n",
    "_Author_: Jerven Bolleman, Principal Software Engineer at SIB\n",
    "\n",
    "_Source_: https://www.linkedin.com/feed/update/activity:7171566017431724033/?trk=feed_main-feed-card_social-actions-comments\n",
    "\n",
    "Here Bolleman et al. show the power of the combination of the IDSM (small molecules - https://lnkd.in/e2JWCd2e) Rhea (https://lnkd.in/e93Pzuva) and UniProt (https://lnkd.in/emDE6CKi) SPARQL endpoints.\n",
    "\n",
    "In the following query we use these three independent #SPARQL endpoints.\n",
    "https://lnkd.in/ejXrwUH3\n",
    "\n",
    "Searching for all human #UniProtKB entries that are annotated to be involved in a disease that are enzymes catalyzing reactions where a substrate or product has Cholestane substructure.\n",
    "\n",
    "We only the #SMILES representation of a Cholestane (https://lnkd.in/e83mrQMr) to start the search. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "66bb8d3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# From UniProt, retrieve all human disease enzymes related to Cholestane metabolite\n",
    "# query (based on Jervern post on LinkedIn)\n",
    "query_disease_enzymes_cholestane = \"\"\"\n",
    "    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n",
    "    PREFIX wikibase: <http://wikiba.se/ontology#>\n",
    "    PREFIX wdt: <http://www.wikidata.org/prop/direct/>\n",
    "    PREFIX wd: <http://www.wikidata.org/entity/>\n",
    "    PREFIX vg: <http://biohackathon.org/resource/vg#>\n",
    "    PREFIX up: <http://purl.uniprot.org/core/>\n",
    "    PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>\n",
    "    PREFIX uberon: <http://purl.obolibrary.org/obo/uo#>\n",
    "    PREFIX taxon: <http://purl.uniprot.org/taxonomy/>\n",
    "    PREFIX sp: <http://spinrdf.org/sp#>\n",
    "    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>\n",
    "    PREFIX sio: <http://semanticscience.org/resource/>\n",
    "    PREFIX sh: <http://www.w3.org/ns/shacl#>\n",
    "    PREFIX schema: <http://schema.org/>\n",
    "    PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>\n",
    "    PREFIX rh: <http://rdf.rhea-db.org/>\n",
    "    PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>\n",
    "    PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "    PREFIX pubmed: <http://rdf.ncbi.nlm.nih.gov/pubmed/>\n",
    "    PREFIX ps: <http://www.wikidata.org/prop/statement/>\n",
    "    PREFIX pq: <http://www.wikidata.org/prop/qualifier/>\n",
    "    PREFIX patent: <http://data.epo.org/linked-data/def/patent/>\n",
    "    PREFIX p: <http://www.wikidata.org/prop/>\n",
    "    PREFIX owl: <http://www.w3.org/2002/07/owl#>\n",
    "    PREFIX orthodbGroup: <http://purl.orthodb.org/odbgroup/>\n",
    "    PREFIX orthodb: <http://purl.orthodb.org/>\n",
    "    PREFIX orth: <http://purl.org/net/orth#>\n",
    "    PREFIX obo: <http://purl.obolibrary.org/obo/>\n",
    "    PREFIX np: <http://nextprot.org/rdf#>\n",
    "    PREFIX nextprot: <http://nextprot.org/rdf/entry/>\n",
    "    PREFIX mnx: <https://rdf.metanetx.org/schema/>\n",
    "    PREFIX mnet: <https://rdf.metanetx.org/mnet/>\n",
    "    PREFIX mesh: <http://id.nlm.nih.gov/mesh/>\n",
    "    PREFIX lscr: <http://purl.org/lscr#>\n",
    "    PREFIX lipidmaps: <https://www.lipidmaps.org/rdf/>\n",
    "    PREFIX keywords: <http://purl.uniprot.org/keywords/>\n",
    "    PREFIX insdcschema: <http://ddbj.nig.ac.jp/ontologies/nucleotide/>\n",
    "    PREFIX insdc: <http://identifiers.org/insdc/>\n",
    "    PREFIX identifiers: <http://identifiers.org/>\n",
    "    PREFIX glyconnect: <https://purl.org/glyconnect/>\n",
    "    PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>\n",
    "    PREFIX genex: <http://purl.org/genex#>\n",
    "    PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n",
    "    PREFIX faldo: <http://biohackathon.org/resource/faldo#>\n",
    "    PREFIX eunisSpecies: <http://eunis.eea.europa.eu/rdf/species-schema.rdf#>\n",
    "    PREFIX ensembltranscript: <http://rdf.ebi.ac.uk/resource/ensembl.transcript/>\n",
    "    PREFIX ensemblterms: <http://rdf.ebi.ac.uk/terms/ensembl/>\n",
    "    PREFIX ensemblprotein: <http://rdf.ebi.ac.uk/resource/ensembl.protein/>\n",
    "    PREFIX ensemblexon: <http://rdf.ebi.ac.uk/resource/ensembl.exon/>\n",
    "    PREFIX ensembl: <http://rdf.ebi.ac.uk/resource/ensembl/>\n",
    "    PREFIX ec: <http://purl.uniprot.org/enzyme/>\n",
    "    PREFIX dcterms: <http://purl.org/dc/terms/>\n",
    "    PREFIX dc: <http://purl.org/dc/terms/>\n",
    "    PREFIX chebislash: <http://purl.obolibrary.org/obo/chebi/>\n",
    "    PREFIX chebihash: <http://purl.obolibrary.org/obo/chebi#>\n",
    "    PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>\n",
    "    PREFIX busco: <http://busco.ezlab.org/schema#>\n",
    "    PREFIX bibo: <http://purl.org/ontology/bibo/>\n",
    "    PREFIX allie: <http://allie.dbcls.jp/>\n",
    "    PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>\n",
    "    PREFIX GO: <http://purl.obolibrary.org/obo/GO_>\n",
    "    PREFIX ECO: <http://purl.obolibrary.org/obo/ECO_>\n",
    "    PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>\n",
    "    SELECT DISTINCT \n",
    "      ?protein \n",
    "      ?disease\n",
    "      ?rhea\n",
    "      ?chebi\n",
    "      ?omim\n",
    "    WHERE {\n",
    "      # find complete ChEBIs with a Cholestane skeleton, via the Czech Elixir node IDSM Sachem \n",
    "      # chemical substructure search.\n",
    "      SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/chebi> {\n",
    "        ?chebi sachem:substructureSearch [\n",
    "            sachem:query\n",
    "            \"[H][C@@]1(CC[C@@]2([H])[C@]3([H])CCC4CCCC[C@]4(C)[C@@]3([H])CC[C@]12C)[C@H](C)CCCC(C)C\"\n",
    "        ] .\n",
    "      }\n",
    "      # Use the fact that UniProt catalytic activities are annotated using Rhea\n",
    "      # Mapping the found ChEBIs o Rhea reactions\n",
    "      SERVICE <https://sparql.rhea-db.org/sparql> {\n",
    "        ?rhea rh:side/rh:contains/rh:compound/rdfs:subClassOf ?chebi .\n",
    "      }\n",
    "      # Match the found Rhea reactions with human UniProtKB proteins\n",
    "      ?protein up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea ;\n",
    "         up:organism taxon:9606 .\n",
    "      # Find only those human entries that have an annotated related disease, and optionally \n",
    "      # map these to OMIM\n",
    "      ?protein up:annotation/up:disease ?disease .\n",
    "      OPTIONAL {\n",
    "            ?disease rdfs:seeAlso ?omim .\n",
    "            ?omim up:database <http://purl.uniprot.org/database/MIM> \n",
    "      }\n",
    "    } \n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4bbb7581",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "      <th>chebi</th>\n",
       "      <th>omim</th>\n",
       "      <th>protein</th>\n",
       "      <th>rhea</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://purl.uniprot.org/diseases/330</td>\n",
       "      <td>http://purl.obolibrary.org/obo/CHEBI_2288</td>\n",
       "      <td>http://purl.uniprot.org/mim/235555</td>\n",
       "      <td>http://purl.uniprot.org/uniprot/P51857</td>\n",
       "      <td>http://rdf.rhea-db.org/46632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://purl.uniprot.org/diseases/330</td>\n",
       "      <td>http://purl.obolibrary.org/obo/CHEBI_2290</td>\n",
       "      <td>http://purl.uniprot.org/mim/235555</td>\n",
       "      <td>http://purl.uniprot.org/uniprot/P51857</td>\n",
       "      <td>http://rdf.rhea-db.org/46640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://purl.uniprot.org/diseases/330</td>\n",
       "      <td>http://purl.obolibrary.org/obo/CHEBI_16074</td>\n",
       "      <td>http://purl.uniprot.org/mim/235555</td>\n",
       "      <td>http://purl.uniprot.org/uniprot/P51857</td>\n",
       "      <td>http://rdf.rhea-db.org/11524</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                disease  \\\n",
       "0  http://purl.uniprot.org/diseases/330   \n",
       "1  http://purl.uniprot.org/diseases/330   \n",
       "2  http://purl.uniprot.org/diseases/330   \n",
       "\n",
       "                                        chebi  \\\n",
       "0  http://purl.obolibrary.org/obo/CHEBI_2288    \n",
       "1  http://purl.obolibrary.org/obo/CHEBI_2290    \n",
       "2  http://purl.obolibrary.org/obo/CHEBI_16074   \n",
       "\n",
       "                                 omim                                 protein  \\\n",
       "0  http://purl.uniprot.org/mim/235555  http://purl.uniprot.org/uniprot/P51857   \n",
       "1  http://purl.uniprot.org/mim/235555  http://purl.uniprot.org/uniprot/P51857   \n",
       "2  http://purl.uniprot.org/mim/235555  http://purl.uniprot.org/uniprot/P51857   \n",
       "\n",
       "                           rhea  \n",
       "0  http://rdf.rhea-db.org/46632  \n",
       "1  http://rdf.rhea-db.org/46640  \n",
       "2  http://rdf.rhea-db.org/11524  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# set the query to be executed against the UniProt endpoint and set the return format to JSON\n",
    "sparql_uniprot.setQuery(query_disease_enzymes_cholestane)\n",
    "sparql_uniprot.setReturnFormat(JSON)\n",
    "\n",
    "NUM_EXAMPLES=3\n",
    "results_disease_enzymes = sparql_uniprot.query().convert()\n",
    "pretty_print(results_disease_enzymes).head(NUM_EXAMPLES)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9354f2cf",
   "metadata": {},
   "source": [
    "(pubchem)=\n",
    "## PubChem\n",
    "\n",
    "Here is a link [to the intro](intro.md). Here is the link to [](pubchem) section. PubChem is a comprehensive knowledge base that focuses its metabolite content on several curated databases such as HMDB. The large amount of data and complexity of PubChem structure make the retrieval of information tailored to the user use case. Once this is clear, PubChem offers a set of tools to help users to retrieve this information in a more systematic way.\n",
    "\n",
    "\n",
    "### Exploring disease pathology\n",
    "\n",
    "#### Retrieval of curated disease-metabolite associations\n",
    "Subset human metabolites from HMDB and cross this data via JOIN or UNION with ChEBI (metabolite role)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a838aa0b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be823a34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "f8e28919",
   "metadata": {},
   "source": [
    "## Other databases \n",
    "\n",
    "We also asked the Plant and Microbiome communities, if there are other databases used to retrieve metabolite-related data. The plant community is focusing mainly on sequence and phenomic data for now. The microbiome community is using KEGG (enzyme sequence list of each microbe).\n",
    "MS data: they don’t know that much about MS databases for microbes. Maybe no standard database. They usually obtain metabolomics data using mass spectrometry such as GC-MS, LC-MS, CE-TOF/MS, etc. Also, \"metabolite-related\" data also includes metabolic pathways and enzymes and the reactions they are involved in; for pathways, KEGG or MetaCyc would work, and for predicting enzymes, DeepGO or a similar system."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c94d5957",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fb10c33",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe46c42f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cacb994f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69fe922e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}